tasktiger/_internal.py

```diff
@@ -57,16 +57,20 @@ def gen_id():
     return binascii.b2a_hex(os.urandom(32)).decode('utf8')
 
 
-def gen_unique_id(serialized_name, args, kwargs):
+def gen_unique_id(queue, serialized_name, args, kwargs):
     """
     Generates and returns a hex-encoded 256-bit ID for the given task name and
     args. Used to generate IDs for unique tasks or for task locks.
     """
+    data = {
+        'func': serialized_name,
+        'args': args,
+        'kwargs': kwargs,
+    }
+    if queue is not None:
+        data['queue'] = queue
```

Comment:
Any reason not to serialize an explicit null?

Reply:
We currently use the same function for to compute the lock ID (in which case we pass `None` for the queue). It would break existing locks if we serialized an explicit null. I can add a comment for this.